_posts/2017-06-06-Spring-IO-2017-The-Spring-ecosystem.md

```diff
@@ -0,0 +1,318 @@
+It's an updated and revised toolkit for cloud-native message-driven microservices. 
+The change was made by Pivotal Software after their experience with Spring Boot. 
+Spring Cloud Data Flow is suitable for processing, analyzing and predicting data. 
+Through streaming it can consume data from an HTTP endpoint and writes the payloads to a database of your choice. 
+It also manages to scale the data pipelines to your liking without any interruptions. 
+After development, an application can be easily executed in Cloud Foundry, Apache YARN, Kubernetes or Apache Mesos, but with the Service Provider Interface you can deploy your application to other runtimes.
+
+## Spring Data
+
+<span class="image right small"><img class="p-image" alt="Spring Data" src="/img/spring-ecosystem/spring-data.png"></span>
+
+Whether you're working with relational or non-relational databases, Spring Data will soothe your needs. 
+As an umbrella project it will ease your way into data access. 
+Some of the related subprojects will help you develop quicker for your favorite database, like Spring Data Mongodb, Spring Data JPA, Spring Data for Apache Cassandra or Spring Data for Apache Solr. 
+And through the help of some community modules this is extended to several others. 
+With Spring Data REST you can expose your Spring Data repository automatically as a REST resource. 
+As usual with Spring projects, they provide an excellent base but can be customised towards your own needs. 
+A full list of subprojects and community projects are available [here](http://projects.spring.io/spring-data).
+
+## Spring HATEOAS
+
+<span class="image left small"><img class="p-image" alt="Spring HATEOAS" src="/img/spring-ecosystem/spring-hateoas.png"></span>
+
+HATEOAS stands for Hypermedia As The Engine Of Application State. 
+It enables the server to update its functionality by decoupling the server and client. 
+With Spring HATEOAS it will be easy to create a REST resource implementation using the HATEOAS as an underlying principle. 
+It will help the client by returning its response in combination with more information on what to do next. 
+If the state of the resource changes, the information on the next steps will also vary throughout the application. 
+HATEOAS requires that the RESTful service methods  GET, POST, PUT, and DELETE or subset are implemented. 
+Spring Security will also help with the relational link creation.
+
+
+## Spring Rest Docs 
+
+<span class="image right small"><img class="p-image" alt="Spring REST Docs" src="/img/spring-ecosystem/spring-rest-docs.png"></span>
+
+When you develop a RESTful service, you'll probably want to document it so it's easy for other developers to implement your API. 
+Spring REST Docs helps you with the documentation process to make it more accurate and readable. 
+It does this by using Asciidoctor which processes text into HTML. 
+Alternatively it can be configured to use Markdown. 
```

Comment:
The advantage here is that **the documentation is always up-to-date with your code**, since the integration tests will fail otherwise.